codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSMiddlewareExecutionGenerator.kt

```diff
@@ -0,0 +1,29 @@
+package software.amazon.smithy.aws.swift.codegen
+
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.config.ConfigProperty
+import software.amazon.smithy.swift.codegen.integration.HTTPProtocolCustomizable
+import software.amazon.smithy.swift.codegen.integration.HttpBindingResolver
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareExecutionGenerator
+import software.amazon.smithy.swift.codegen.middleware.OperationMiddleware
+
+class AWSMiddlewareExecutionGenerator(
+    ctx: ProtocolGenerator.GenerationContext,
+    private val writer: SwiftWriter,
+    httpBindingResolver: HttpBindingResolver,
+    httpProtocolCustomizable: HTTPProtocolCustomizable,
+    operationMiddleware: OperationMiddleware,
+    operationStackName: String
+) : MiddlewareExecutionGenerator(
+    ctx, writer, httpBindingResolver, httpProtocolCustomizable, operationMiddleware, operationStackName
+) {
+
+    override fun renderConfigPropertyToContext(configProperty: ConfigProperty) {
+        when (configProperty.name) {
+            "accountIdEndpointMode" -> { // Used for business metrics
+                writer.write("  .withAccountIDEndpointMode(value: config.accountIdEndpointMode)")
+            }
+        }
+    }
+}
```

Comment:
Wrote comment in smithy-swift PR as well; this is redundant; we can use existing extension point for printing custom AWS specific context attributes. See [here](https://github.com/awslabs/aws-sdk-swift/blob/b4a7d85261a5315db5f66f5c90dbc9646e13595c/codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSHTTPProtocolCustomizations.kt#L27-L38). The extension point used by previous link is [here](https://github.com/smithy-lang/smithy-swift/blob/fad1c76d242539f1ee5e29817ce720d45b84583c/smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt#L118).